[RF] Deprecate RooSetPair class by moving it to RooFitLegacy

guitargeek opened the pull request:
The RooSetPair is not used in non-legacy RooFit code anymore, and it can easily be replaced by a `std::pair<RooArgSet*,RooArgSet*>`.

hageboeck commented on roofit/roofitcore/src/RooFitLegacy/RooHashTable.cxx:
```diff
@@ -32,7 +32,7 @@ ClassImp(RooHashTable);
 \file RooHashTable.cxx
 \class RooHashTable
-\ingroup Roofitcore
+\ingroup Roofitlegacy
```
Did you define this group? Doxygen needs an explicit
```
/// \group Roofitlegacy
/// Description 1st sentence.
/// Description ...
```
to make something reasonable of it.

guitargeek replied:
It seems the group is already defined: https://root.cern/doc/master/group__Roofitlegacy.html

hageboeck commented on roofit/roofitcore/inc/RooFitLegacy/RooSetPair.h:
```diff
@@ -20,6 +20,8 @@
+#include <ROOT/RConfig.hxx>
+
 class RooLinkedListElem ;
 class TBuffer ;
```
```suggestion
```

hageboeck commented on roofit/roofitcore/inc/RooFitLegacy/RooSetPair.h:
```diff
@@ -20,6 +20,8 @@
 #include "TObject.h"
 #include "RooArgSet.h"
```
```suggestion
class RooArgSet;
```

hageboeck reviewed (approved):
LGTM in principle, but note that code that uses this won't compile, since the header is missing now.
Maybe you can mention in the release notes how to find the header.

hageboeck commented:
Also @guitargeek I edited out some includes. There is a second commit now.

guitargeek commented:
> Also @guitargeek I edited out some includes. There is a second commit now.

Thanks for that! I fixed up my own commit and also squashed your include fix on top of it.
